packages/tinybased/src/lib/search/searched-react.spec.tsx

```diff
@@ -0,0 +1,135 @@
+import { SearchParams, SearchResult } from '@lyrasearch/lyra';
+import { renderHook } from '@testing-library/react-hooks';
+import {
+  notesTable,
+  NOTE_1,
+  NOTE_2,
+  NOTE_3,
+  usersTable,
+  USER_1,
+  USER_2,
+  USER_3,
+  USER_4,
+  USER_5,
+} from '../../fixture/database';
+import { waitAMoment } from '../../testing/utils';
+import { SchemaBuilder } from '../SchemaBuilder';
+import { ObjectToCellStringType } from '../TableBuilder';
+import { InferSchema, OnlyStringKeys } from '../types';
+import { generateSearched } from './generateSearched';
+
+const schemaBuilderTypeReference = () =>
+  new SchemaBuilder()
+    .addTable(usersTable)
+    .addTable(notesTable)
+    .defineHydrators({
+      users: () => Promise.resolve([USER_1, USER_2, USER_3, USER_4, USER_5]),
+      notes: () => Promise.resolve([NOTE_1, NOTE_2, NOTE_3]),
+    });
+
+describe('Searched in react', () => {
+  let schemaBuilder!: ReturnType<typeof schemaBuilderTypeReference>;
+  let tb!: Awaited<ReturnType<(typeof schemaBuilder)['build']>>;
+
+  let useSearch!: <
```

Comment:
Wow that is quite the type signature. Was it not possible to just infer it based on `typeof useSearch`?